### PR TITLE
Fix NONE resolution ServiceEntry

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/25844

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -1079,27 +1080,34 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(node *model.
...
-								configgen.buildSidecarOutboundListenerForPortOrUDS(node, listenerOpts, pluginParams, listenerMap,
-									virtualServices, actualWildcard)
+						instances, err := push.InstancesByPort(service, servicePort.Port, nil)
+						if err != nil {
+							// This is impossible, see https://github.com/istio/istio/pull/26616 which removes the error
+							log.Errorf("instances by port got error: %v", err)
+							continue
+						}
+						if service.Attributes.ServiceRegistry != string(serviceregistry.Kubernetes) && len(instances) == 0 {
+							// A Kubernetes service with no endpoints means there are no endpoints at all, so don't both sending.
```
Comment (mandarjog) [inline pilot/pkg/networking/core/v1alpha3/listener.go:1090]: nit: don't bother sending ?

**Comment by mandarjog** [comment]
Can we add an e2e test for this? or someone needs to manually verify the scenario.

**Comment by mandarjog** [comment]
cc @rshriram

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
feel free to remove the hold once the typo addressed

**Comment by howardjohn** [comment]
I removed the hold since this is time sensitive and just for a comment - I will update tomorrow

**Pull request merged**